src/Components/CanvasComponent.tsx

```diff
@@ -1,24 +1,29 @@
 import React from "react";
-import { SketchPicker } from "react-color";
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faEraser, faCircle, faSquare, faPaintBrush, faDownload, faSync } from '@fortawesome/free-solid-svg-icons'
 
 interface Point {
   x: number;
   y: number;
 }
 
-interface Color {
-  hex: string;
-}
 
 class CanvasComponent extends React.Component {
+  colorInput;
+
+  constructor(props) {
+    super(props);
+    this.colorInput = React.createRef();
+  }
+
   state = {
-    background: "#121212",
+    color: "#000000",
     choice: "brush",
     canvas: {} as HTMLCanvasElement,
```

Comment:
Canvas should not be a state, it should be a ref.

Reply:
Basically, whatever data you need to render the component (color, choice, isDrawStarted, etc.) should either be a state or prop. UI elements like canvas, websocket connection, etc are important but they are not exactly data that you want to display. They should be stored inside refs. They are also mutable which might cause undefined behaviour with state: state and props should always be immutable.